### PR TITLE
copr: support dividde

### PR DESCRIPTION
**Pull request by BusyJay**
@ngaut @coocood @shenli @qiuyesuifeng PTAL

### CODE REVIEW

**Comment by coocood** [comment]
LGTM

**Comment by BusyJay** [comment]
@ngaut @shenli @qiuyesuifeng @siddontang PTAL

File: src/util/codec/mysql/decimal.rs [hunk 1]
```diff
@@ -184,6 +195,22 @@ fn sub(lhs: u32, rhs: u32, carry: &mut i32, res: &mut u32) {
     }
 }
 
+/// `sub2` substracts rhs and carry from lhs, stores the diff and new carry.
+/// the new carry may be 2.
+fn sub2(lhs: u32, rhs: u32, carry: &mut i32, res: &mut u32) {
+    let mut diff = lhs as i32 - rhs as i32 - *carry;
+    if diff < -(WORD_BASE as i32) {
```
Comment (shenli) [inline src/util/codec/mysql/decimal.rs]: This is not the same with TiDB decimal. https://github.com/pingcap/tidb/blob/master/mysql/mydecimal.go#L91

Comment (shenli) [inline src/util/codec/mysql/decimal.rs]: I see.

**Comment by shenli** [comment]
LGTM
But I suggest some guy familiar with rust take a look at this PR.

**Comment by BusyJay** [comment]
@siddontang @disksing PTAL

**Comment by siddontang** [comment]
LGTM

**Pull request merged**